app/lib/action_controller/request_forgery_protection/exception_and_reset_strategy.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module ActionController
+  module RequestForgeryProtection
+    class ExceptionAndResetStrategy
+      def initialize(controller)
+        @controller = controller
+      end
+
+      def handle_unverified_request
+        @controller.reset_session
+        exception.handle_unverified_request
+      end
+
+      private
+
+      def exception
+        @exception || ActionController::RequestForgeryProtection::ProtectionMethods::Exception.new(@controller)
```

Comment:
```suggestion
        @exception ||= ActionController::RequestForgeryProtection::ProtectionMethods::Exception.new(@controller)
```

Or was `||` intentional?